cel/env.go

```diff
@@ -138,6 +171,47 @@ func (e *Env) Check(ast *Ast) (*Ast, *Issues) {
 		typeMap: res.GetTypeMap()}, nil
 }
 
+// Compile combines the Parse and Check phases CEL program compilation to produce an Ast and
+// associated issues.
+//
+// If an error is encountered during parsing the Compile step will not continue with the Check
+// phase. If non-error issues are encountered during Parse, they may be combined with any issues
+// discovered during Check.
+//
+// Note, for parse-only uses of CEL use Parse.
+func (e *Env) Compile(txt string) (*Ast, *Issues) {
+	return e.CompileSource(common.NewTextSource(txt))
+}
+
+// CompileSource combines the Parse and Check phases CEL program compilation to produce an Ast and
+// associated issues.
+//
+// If an error is encountered during parsing the CompileSource step will not continue with the
+// Check phase. If non-error issues are encountered during Parse, they may be combined with any
+// issues discovered during Check.
+//
+// Note, for parse-only uses of CEL use Parse.
+func (e *Env) CompileSource(src common.Source) (*Ast, *Issues) {
+	ast, iss := e.ParseSource(src)
+	if iss != nil && iss.Err() != nil {
```

Comment:
what do you think about changing how method `Err()` on issues works? maybe it could work is similar way as grpc getters.
if issues is nil, return nil.

Reply:
I hear you, and while there really isn't any code to show why there is an `Issues` object instead of an `errors` object, it's because we intend to surface deprecation notices through the `Issues` object. There are a couple upcoming standard library changes which will likely warrant a deprecation noticed, but basically even if you're using something deprecated it will still work, it'll just point out that you're using something deprecated and should do something else instead.

Reply:
@TristonianJones I am not sure if you understand me well. 
I understand that you want to return typed Issues. 
However I still think that checking issues could be simplified:
```
func (i *Issues) Err() error {
	if i == nil {
		return nil
	}
	if len(i.errs.GetErrors()) > 0 {
		return errors.New(i.errs.ToDisplayString())
	}
	return nil
}
```
using that approach consumers of API could simply check
```
if iss.Err() != nil {}
```

Reply:
Ah, okay, I get it now. I was unfortunately over-indexed on prior feedback of "why not just an error?". I can return an empty issues list when there are no errors. I'll write a PR for it later this week hopefully.